In [ ]:
import pandas as pd

# Task 1

## 1. CAC (Customer Acquisition Cost)

### 1.1. Monthly expenses

In [ ]:
# Load data
monthly_expense_gs_id = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
monthly_expense_url = f'https://docs.google.com/spreadsheets/d/{monthly_expense_gs_id}/export?format=xlsx'

monthly_expense_df = pd.read_excel(monthly_expense_url, sheet_name='Sheet1')
monthly_expense_df

#      month           category                  item  amount
0    1 2023-01-01       Server Costs           AWS Hosting    8000
1    2 2023-01-01       Server Costs  Google Cloud Storage    4000
2    3 2023-01-01  Software Licenses        Atlassian Jira    1200
3    4 2023-01-01  Software Licenses                 Slack     800
4    5 2023-01-01  Software Licenses            Salesforce    1500
5    6 2023-01-01  Software Licenses                  Zoom     500
6    7 2023-01-01      Office Rental           Office Rent   10000
7    8 2023-01-01              Other       Office Supplies     500
8    9 2023-01-01              Other       Travel Expenses    3000
9   10 2023-02-01       Server Costs           AWS Hosting    8200
10  11 2023-02-01       Server Costs  Google Cloud Storage    4200
11  12 2023-02-01  Software Licenses        Atlassian Jira    1300
12  13 2023-02-01  Software Licenses                 Slack     850
13  14 2023-02-01  Software Licenses            Salesforce    1600
14  15 2023-02-01  Software Licenses                  Zoom     520
15  16 2023-02-01      Office Rental           Office Rent   10000
16  17 2023-02-01              Other       Office Supplies     550
17  18 2023-02-01              Other       Travel Expenses    3100
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
monthly_expense_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   #         27 non-null     int64         
 1   month     27 non-null     datetime64[ns]
 2   category  27 non-null     object        
 3   item      27 non-null     object        
 4   amount    27 non-null     int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 1.2+ KB


In [ ]:
# Filter last month data
max_month_expense = monthly_expense_df['month'].max()

last_month_expense_df = monthly_expense_df[monthly_expense_df['month'] == max_month_expense]
last_month_expense_df.head()

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700

In [ ]:
# Calculate monthly expense cost
crm_cost = last_month_expense_df[last_month_expense_df['item'] == 'Salesforce']['amount'].values[0]
crm_cost

np.int64(1700)

### 1.2. Payroll

In [ ]:
# Load data
payroll_gs_id = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
payroll_url = f'https://docs.google.com/spreadsheets/d/{payroll_gs_id}/export?format=xlsx'

payroll_df = pd.read_excel(payroll_url, sheet_name='Sheet1')
payroll_df

month   department   employee_name            position  paid
0  2023-01-01        Sales        John Doe       Sales Manager  1500
1  2023-01-01        Sales      Jane Smith     Sales Associate   600
2  2023-01-01        Sales       Jim Brown     Sales Associate   700
3  2023-01-01        Sales    Laura Miller     Sales Associate   800
4  2023-01-01    Marketing   Alice Johnson   Marketing Manager  1650
5  2023-01-01    Marketing       Bob Davis  Content Specialist   700
6  2023-01-01  Engineering  Charlie Wilson    Senior Developer  2500
7  2023-01-01  Engineering       Diana Lee           Developer  1700
8  2023-01-01  Engineering       Eva White    Junior Developer  1000
9  2023-01-01           HR     Frank Green          HR Manager  1300
10 2023-01-01           HR      Gary Black       HR Specialist   700
11 2023-01-01      Support    Hannah Scott        Support Lead   600
12 2023-01-01      Support      Ian Harris  Support Specialist   500
13 2023-01-01      Support     Emily Clark  Support Specialist   450
14 2023-01-01    Analytics   Michael Brown   Analytics Manager  1700
15 2023-01-01    Analytics    Sarah Wilson        Data Analyst  1200
16 2023-01-01    Analytics       David Lee        Data Analyst  1300
17 2023-02-01        Sales        John Doe       Sales Manager  1500
18 2023-02-01        Sales      Jane Smith     Sales Associate   600
19 2023-02-01        Sales       Jim Brown     Sales Associate   700
20 2023-02-01        Sales    Laura Miller     Sales Associate  5100
21 2023-02-01    Marketing   Alice Johnson   Marketing Manager  1650
22 2023-02-01    Marketing       Bob Davis  Content Specialist   700
23 2023-02-01  Engineering  Charlie Wilson    Senior Developer  2500
24 2023-02-01  Engineering       Diana Lee           Developer  1700
25 2023-02-01  Engineering       Eva White    Junior Developer  1000
26 2023-02-01           HR     Frank Green          HR Manager  1300
27 2023-02-01           HR      Gary Black       HR Specialist   700
28 2023-02-01      Support    Hannah Scott        Support Lead   600
29 2023-02-01      Support      Ian Harris  Support Specialist   500
30 2023-02-01      Support     Emily Clark  Support Specialist   450
31 2023-02-01    Analytics   Michael Brown   Analytics Manager  1700
32 2023-02-01    Analytics    Sarah Wilson        Data Analyst  1200
33 2023-02-01    Analytics       David Lee        Data Analyst  1300
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

In [ ]:
# Filter last month data
last_month_payroll_df = payroll_df[payroll_df['month'] == max_month_expense]
last_month_payroll_df.head()

month department  employee_name           position  paid
34 2023-03-01      Sales       John Doe      Sales Manager  1500
35 2023-03-01      Sales     Jane Smith    Sales Associate   600
36 2023-03-01      Sales      Jim Brown    Sales Associate   700
37 2023-03-01      Sales   Laura Miller    Sales Associate   800
38 2023-03-01  Marketing  Alice Johnson  Marketing Manager  1650

In [ ]:
# Calculate sales and marketing payroll cost
last_month_sales_marketing_payroll_cost = last_month_payroll_df[last_month_payroll_df['department'].isin(['Sales', 'Marketing'])]['paid'].sum()
last_month_sales_marketing_payroll_cost

np.int64(5950)

### 1.3. Marketing spendings

In [ ]:
# Load data
marketing_spendings_gs_id = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
marketing_spendings_url = f'https://docs.google.com/spreadsheets/d/{marketing_spendings_gs_id}/export?format=xlsx'

marketing_spendings_df = pd.read_excel(marketing_spendings_url, sheet_name='Sheet1')
marketing_spendings_df

date       channel  spending
0   2023-01-01    Google Ads       784
1   2023-01-01  Facebook Ads       659
2   2023-01-01  LinkedIn Ads       729
3   2023-01-01   Twitter Ads       292
4   2023-01-02    Google Ads       935
..         ...           ...       ...
355 2023-03-30   Twitter Ads       960
356 2023-03-31    Google Ads       143
357 2023-03-31  Facebook Ads       183
358 2023-03-31  LinkedIn Ads       533
359 2023-03-31   Twitter Ads       909

[360 rows x 3 columns]

In [ ]:
target_month = max_month_expense.to_period('M')
target_month

Period('2023-03', 'M')

In [ ]:
marketing_spendings_df['date'].dt.to_period('M')

0      2023-01
1      2023-01
2      2023-01
3      2023-01
4      2023-01
        ...   
355    2023-03
356    2023-03
357    2023-03
358    2023-03
359    2023-03
Name: date, Length: 360, dtype: period[M]

In [ ]:
# Calculate marketing spending cost
marketing_spendings_cost = marketing_spendings_df[marketing_spendings_df['date'].dt.to_period('M') == target_month]['spending'].sum()

In [ ]:
# Calculate total sales and marketing expenses
total_sales_marketing_expense = crm_cost + last_month_sales_marketing_payroll_cost + marketing_spendings_cost
total_sales_marketing_expense

np.int64(76480)

### 1.4. Number of customers

In [ ]:
# Load data
receipt_hs_gs_id = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
receipt_url = f'https://docs.google.com/spreadsheets/d/{receipt_hs_gs_id}/export?format=xlsx'

receipt_df = pd.read_excel(receipt_url, sheet_name='Sheet1')
receipt_df

date  customer_id  receipt_amount  new_customer
0   2023-01-01         2653              67             1
1   2023-01-01         2731             271             1
2   2023-01-01         1277             231             0
3   2023-01-01         2094             107             0
4   2023-01-01         1314             416             0
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[929 rows x 4 columns]

In [ ]:
last_month_receipt_df = receipt_df[receipt_df['date'].dt.to_period('M') == target_month]
last_month_receipt_df

date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

In [ ]:
new_customer_df = last_month_receipt_df[last_month_receipt_df['new_customer'] == 1]
num_new_customer = len(new_customer_df)
num_new_customer

63

### 1.5. Calculate CAC

In [ ]:
CAC = total_sales_marketing_expense / num_new_customer
CAC

np.float64(1213.968253968254)

## 2. ARPU (Average Revenue Per User)

### 2.1. Calculate total revenue

In [ ]:
total_revenue = last_month_receipt_df['receipt_amount'].sum()
total_revenue

np.int64(83033)

### 2.2. Count number of users

In [ ]:
num_users = len(last_month_receipt_df['customer_id'].unique())
num_users

292

### 2.3. Calculate ARPU

In [ ]:
ARPU = total_revenue / num_users
ARPU

np.float64(284.3595890410959)

## 3. COGS (Cost of Goods Sold)

### 3.1. Calculate production cost

In [ ]:
last_month_expense_df

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
last_month_expense_df[last_month_expense_df['category'].isin(['Server Costs','Software Licenses'])]

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540

In [ ]:
work_items = ['AWS Hosting','Google Cloud Storage','Atlassian Jira']
shared_items = ['Slack', 'Zoom']

#Assume 60% cost for Slack and Zoom is allocated to COGS
work_items_expense = last_month_expense_df[last_month_expense_df['item'].isin(work_items)]['amount'].sum()
shared_items_expense = last_month_expense_df[last_month_expense_df['item'].isin(shared_items)]['amount'].sum()

server_n_software_expense = work_items_expense + shared_items_expense * 0.6
server_n_software_expense

np.float64(15064.0)

### 3.2. Calculate production payroll

In [ ]:
production_payroll = last_month_payroll_df[last_month_payroll_df['department'] == 'Engineering']['paid'].sum()
production_payroll

np.int64(5200)

### 3.3. Calculate COGS

In [ ]:
COGS = server_n_software_expense + production_payroll
COGS

np.float64(20264.0)

## 4. Gross margin

In [ ]:
Gross_margin = (total_revenue - COGS) / total_revenue *100
Gross_margin

np.float64(75.5952452639312)

## 5. LTV (Customer Lifetime Value)

In [ ]:
# Load data
customer_lifespan_gs_id = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
customer_lifespan_url = f'https://docs.google.com/spreadsheets/d/{customer_lifespan_gs_id}/export?format=xlsx'

lifespan_df = pd.read_excel(customer_lifespan_url, sheet_name='Sheet1')
lifespan_df

Unnamed: 0 start_date churn_date
0         1000 2021-11-15 2022-09-14
1         1001 2022-04-15 2023-02-16
2         1002 2022-10-30 2023-02-04
3         1003 2021-08-22 2023-02-07
4         1004 2021-08-23 2022-02-02
..         ...        ...        ...
95        1095 2023-01-09 2023-02-28
96        1096 2020-07-08 2020-08-09
97        1097 2020-10-18 2022-05-08
98        1098 2020-06-17 2022-06-13
99        1099 2020-06-25 2023-02-12

[100 rows x 3 columns]

In [ ]:
lifespan_df['lifespan_days'] = (lifespan_df['churn_date'] - lifespan_df['start_date']).dt.days
lifespan_df

Unnamed: 0 start_date churn_date  lifespan_days
0         1000 2021-11-15 2022-09-14            303
1         1001 2022-04-15 2023-02-16            307
2         1002 2022-10-30 2023-02-04             97
3         1003 2021-08-22 2023-02-07            534
4         1004 2021-08-23 2022-02-02            163
..         ...        ...        ...            ...
95        1095 2023-01-09 2023-02-28             50
96        1096 2020-07-08 2020-08-09             32
97        1097 2020-10-18 2022-05-08            567
98        1098 2020-06-17 2022-06-13            726
99        1099 2020-06-25 2023-02-12            962

[100 rows x 4 columns]

In [ ]:
# Calculate the average lifespan in days
avg_lifespan_days = lifespan_df['lifespan_days'].mean()

# Convert the average lifespan to months
avg_lifespan_months = avg_lifespan_days / 30
avg_lifespan_months

np.float64(9.841333333333333)

In [ ]:
# Calculate LTV
LTV = ARPU * avg_lifespan_months * Gross_margin
LTV

np.float64(211551.59315068493)

## 6. LTV / CAC

In [ ]:
LTV/CAC

np.float64(174.26451841648995)

# Task 2

## Conclusion

The company demonstrates exceptionally strong unit economics. With a **Customer Acquisition Cost (CAC) of 1,214** and a **Lifetime Value (LTV) of 211,552**, the **LTV/CAC ratio is 174.3**, which is far above the industry benchmark (typically a healthy ratio is 3–5). This indicates that each acquired customer generates significant long-term profitability relative to the cost of acquisition.
The **ARPU (284.36)** combined with a **gross margin of 75.6%** suggests that the company retains a high share of revenue after direct costs, reinforcing financial efficiency. Despite relatively high **COGS (20,264)**, the profitability potential per customer remains strong given the magnitude of LTV.
Overall, the company is in an excellent financial position from a unit economics perspective, with a scalable and profitable customer model.

## Recommendations

### 1. Scale Customer Acquisition



*   With such a high LTV/CAC ratio, the company has ample room to increase marketing and sales investment to acquire more customers, as each acquisition pays off many times over.
*   Carefully track marginal CAC as marketing spend grows to ensure efficiency remains high.




### 2. Optimize Pricing and Monetization



*   Given the strong ARPU and gross margins, consider testing premium pricing tiers, upselling, or cross-selling to maximize per-customer revenue.
*   Explore subscription or retention-based models to further lock in long-term value.



### 3. Sustain Retention Efforts



*   The high LTV suggests long customer lifespans. Continue to invest in customer support, product quality, and engagement initiatives to maintain or extend retention.
*   Implement loyalty programs or data-driven personalization to strengthen stickiness.



### 4. Monitor Cost Structure



*   While COGS is significant in absolute terms, margins remain strong. Keep monitoring supply chain, operational efficiency, and vendor contracts to prevent erosion of profitability.



### 5. Plan for Scalable Growth



*   With such favorable unit economics, focus on sustainable scaling: ensuring operational capacity, maintaining customer experience quality, and reinvesting profits into expansion opportunities.

